From 39a25823bc56d30165f5c04d83c2d51053cf297e Mon Sep 17 00:00:00 2001
From: Sai Surya Teja <sssteja@habana.ai>
Date: Tue, 05 Jul 2022 09:18:28 +0300
Subject: [PATCH] [SW-93024] gaudi2: clear pcie dbi msix address match registers

PCIE_DBI_MSIX_ADDRESS_MATCH_LOW/HIGH_OFF registers are configured in
arc-mgmt based on a LKD cpucp command to direct the MSIX interrupts to
HBW address space as a workaround for exisitng issues with LBW based
MSIX.

This patch clears the above registers in the ppboot to ensure the MSIX
are not enabled by default until FW receives a cpucp command from LKD.

Change-Id: I9c20ed7afa9f97a41cb67d8089de13fd4e3dda6f
---

diff --git a/arch/gaudi2/arc/common.c b/arch/gaudi2/arc/common.c
index 98f6c87..3c88fac 100644
--- a/arch/gaudi2/arc/common.c
+++ b/arch/gaudi2/arc/common.c
@@ -203,6 +203,12 @@
 	init_arc_mem(0, ARC0_ICCM_FULL_BASE, ARC0_ICCM_SIZE);
 }
 
+static void gaudi2_clear_msix_addr_match(void)
+{
+	regmap_writel(0, mmPCIE_DBI_MSIX_ADDRESS_MATCH_HIGH_OFF);
+	regmap_writel(0, mmPCIE_DBI_MSIX_ADDRESS_MATCH_LOW_OFF);
+}
+
 void hl_plat_early_init(void)
 {
 	/* Functionality used to be done in BTL */
@@ -216,6 +222,8 @@
 
 	gaudi2_axi_drain_ena();
 
+	gaudi2_clear_msix_addr_match();
+
 	/* set reset source */
 	hl_reset_src_init();
 
